### PR TITLE
Add version constraints of `setuptools`

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

Fix CI.

## Description of the changes

Same as https://github.com/optuna/optuna/pull/3207.

### CODE REVIEW

**Comment by nzw0301** [comment]
Dear reviewers, to fix this issue, we have at least two ways:

1. Add version constraint of `setuptools`: Optuna's main repo takes this approach. But `setuptools` is a general package, so this constraint might be removed in the near future.
2. Use a workaround as https://github.com/optuna/optuna-examples/pull/76/commits/3c937c9a0c581a5074a0b952d526dd8100a51635 by following https://github.com/mpi4py/mpi4py/issues/157#issuecomment-1001022274.

File: .github/workflows/pytorch.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
         pip install git+https://github.com/optuna/optuna.git
         python -c 'import optuna'
 
-        pip install -r pytorch/requirements.txt
+        SETUPTOOLS_USE_DISTUTILS=stdlib pip install -r pytorch/requirements.txt
```
Comment (not522) [inline .github/workflows/pytorch.yml:36]: Could you add a TODO comment for this line?

Comment (nzw0301) [reply .github/workflows/pytorch.yml]: Thank you for your comment. I undo the last commit for consistency with the optuna repo's update.

**Review by hvy — APPROVED** [review]
LGTM!

Discussed with @nzw0301 outside this PR, to keep the fix consistent with the main optuna/optuna repo. Note that this is a temporary fix to be reverted after the next release of mpi4py.

File: .github/workflows/pytorch.yml (hunk 1 above)
Comment (hvy) [reply .github/workflows/pytorch.yml]: Let me merge this PR as it is now since it's fixing the currently broken daily CI. 🙇

**Pull request merged**